src/App/App.js

```diff
@@ -57,71 +33,17 @@ class App extends Component {
 
     this.setState({ ...this.state, error: 'Looks like we where unable to find that title.' });
   }
-
-  viewMovie = (id) => {
-    const movie = this.findMovieById(id);
-    if(movie) {
-      this.setState({
-            ...this.state, 
-            isMovieView: true,
-            error: '',
-            movie: {
-            id: movie.id,
-            title: movie.title,
-            posterPath: movie['poster_path'],
-            backdropPath: movie['backdrop_path'],
-            releaseDate: movie['release_date'],
-            overview: movie.overview,
-            genres: movie.genres,
-            budget: movie.budget,
-            revenue: movie.revenue,
-            runtime: movie.runtime,
-            tagline: movie.tagline,
-            averageRating: movie['average_rating']
-          }})
-          
-          return;
-    }
-
-    getMovie(id)
-      .then(response => response.json())
-      .then(data => {
-        this.setState({...this.state, 
-          isMovieView: true,
-          fetchedMovies: [...this.state.fetchedMovies, data.movie],
-          error: '',
-          movie: {
-          id: data.movie.id,
-          title: data.movie.title,
-          posterPath: data.movie['poster_path'],
-          backdropPath: data.movie['backdrop_path'],
-          releaseDate: data.movie['release_date'],
-          overview: data.movie.overview,
-          genres: data.movie.genres,
-          budget: data.movie.budget,
-          revenue: data.movie.revenue,
-          runtime: data.movie.runtime,
-          tagline: data.movie.tagline,
-          averageRating: data.movie['average_rating']
-        }})
-      })
-      .catch(err =>{
-        console.log(err)
-        this.setState({ ...this.state, error: 'Looks like something went wrong.' })
-      })
-  }
  
```

Comment:
Moved everything that handles fetching a movie to the Movie component, which is now a class component